state.go

```diff
@@ -257,6 +258,37 @@ func (m *Memberlist) probeNode(node *nodeState) {
 		}
 	}
 
+	// Also make an attempt to contact the node directly over TCP. This
+	// helps prevent confused clients who get isolated from UDP traffic
+	// but can still speak TCP (which also means they can possibly report
+	// misinformation to other nodes via anti-entropy), avoiding flapping in
+	// the cluster.
+	fallbackCh := make(chan bool)
+	if node.PMax >= 3 {
+		destAddr := &net.TCPAddr{IP: node.Addr, Port: int(node.Port)}
```

Comment:
Haha - I had the bool and took it out because it felt weird to have that extra interface just for that behavior. After sleeping on it, though, I agree with you it would simplify things on the side where things are already complicated, and it'll get rid of the `dial` call outside of the `net` module - I'll change it.